### PR TITLE
[Silabs] Update Docker for SLC

### PR DESCRIPTION
**Pull request by jepenven-silabs**
Update GSDK inside docker container and add SLC for Silabs specific code generation

### CODE REVIEW

File: integrations/docker/images/chip-build-efr32/Dockerfile [hunk 1]
```diff
@@ -7,7 +7,32 @@ RUN set -x \
...
     && apt-get clean \
     && rm -rf /var/lib/apt/lists/ \
     && : # last line
+
+
+#Clone Gecko SDK 4.2.0 (d4854d2)
+RUN git clone https://github.com/SiliconLabs/gecko_sdk.git && \
+    cd ./gecko_sdk && \
+    git fetch origin && \
+    git checkout d4854d2
```
Comment (andy31415) [inline integrations/docker/images/chip-build-efr32/Dockerfile:22]: should there be a tag for this? then maybe something like `git clone --depth=1 --branch=... ...` would result in a smaller image and shorter instructions.

File: integrations/docker/images/chip-build-efr32/Dockerfile [hunk 2]
```diff
@@ -7,7 +7,32 @@ RUN set -x \
...
+
+ENV GSDK_ROOT=/gecko_sdk/
+
+# SLC-cli install
+RUN wget https://www.silabs.com/documents/login/software/slc_cli_linux.zip && \
+    unzip ./slc_cli_linux.zip -d ./ && \
+    cd ./slc_cli && \
+    pip3 install -r ./requirements.txt
```
Comment (andy31415) [inline integrations/docker/images/chip-build-efr32/Dockerfile:30]: should we dleete the zip file after install so that the image is smaller?

**Review by andy31415 — COMMENTED** [review]
@jepenven-silabs please also update version file

**Review by bzbarsky-apple — APPROVED** [review]

File: integrations/docker/images/chip-build-efr32/Dockerfile (hunk 1 above)
Comment (jepenven-silabs) [reply integrations/docker/images/chip-build-efr32/Dockerfile]: Done

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**